packages/engine/crawlers/solana/collectibles.ts

```diff
@@ -90,12 +96,14 @@ export const addCollectibleToState = async (
 			},
 			count: 1,
 		});
-	} else {
-		collection.count += 1;
+	} else if (!collectible) {
+		collectibleActions.updateCollectionAmount(
+			collectionId,
+			(collection.count += 1),
+		);
 	}
 
-	const collectibleId = `${address}/${nft.mint.address.toString()}`;
-	collectibleState.map.set(collectibleId, {
+	collectibleActions.setStorageCollectible(collectibleId, {
```

Comment:
I think storage is in data level, not in the app state level, so just make it a single function instead of use as a state action